src/diffusers/pipelines/latent_consistency_models/pipeline_latent_consistency_models.py

```diff
@@ -0,0 +1,684 @@
+                following arguments: `callback(step: int, timestep: int, latents: torch.FloatTensor)`.
+            callback_steps (`int`, *optional*, defaults to 1):
+                The frequency at which the `callback` function is called. If not specified, the callback is called at
+                every step.
+            cross_attention_kwargs (`dict`, *optional*):
+                A kwargs dictionary that if specified is passed along to the [`AttentionProcessor`] as defined in
+                [`self.processor`](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/attention_processor.py).
+            guidance_rescale (`float`, *optional*, defaults to 0.0):
+                Guidance rescale factor from [Common Diffusion Noise Schedules and Sample Steps are
+                Flawed](https://arxiv.org/pdf/2305.08891.pdf). Guidance rescale factor should fix overexposure when
+                using zero terminal SNR.
+            clip_skip (`int`, *optional*):
+                Number of layers to be skipped from CLIP while computing the prompt embeddings. A value of 1 means that
+                the output of the pre-final layer will be used for computing the prompt embeddings.
+
+        Returns:
+            [`~pipelines.stable_diffusion.StableDiffusionPipelineOutput`] or `tuple`:
+                If `return_dict` is `True`, [`~pipelines.stable_diffusion.StableDiffusionPipelineOutput`] is returned,
+                otherwise a `tuple` is returned where the first element is a list with the generated images and the
+                second element is a list of `bool`s indicating whether the corresponding generated image contains
+                "not-safe-for-work" (nsfw) content.
+        """
+        # 0. Default height and width to unet
+        height = height or self.unet.config.sample_size * self.vae_scale_factor
+        width = width or self.unet.config.sample_size * self.vae_scale_factor
+
+        # 1. Check inputs. Raise error if not correct
+        self.check_inputs(prompt, height, width, callback_steps, prompt_embeds)
+
+        # 2. Define call parameters
+        if prompt is not None and isinstance(prompt, str):
+            batch_size = 1
+        elif prompt is not None and isinstance(prompt, list):
+            batch_size = len(prompt)
+        else:
+            batch_size = prompt_embeds.shape[0]
+
+        device = self._execution_device
+        # do_classifier_free_guidance = guidance_scale > 0.0  # In LCM Implementation:  cfg_noise = noise_cond + cfg_scale * (noise_cond - noise_uncond) , (cfg_scale > 0.0 using CFG)
+
+        # 3. Encode input prompt
+        lora_scale = cross_attention_kwargs.get("scale", None) if cross_attention_kwargs is not None else None
+
+        prompt_embeds, _ = self.encode_prompt(
+            prompt,
+            device,
+            num_images_per_prompt,
+            False,  # Don't need to get negative prompts due to LCM guided distillation
```

Comment:
To clarify: LCM currently supports only the unconditional prompt "" and does not support any other types of negative prompts.

Reply:
I see. So if I understand correctly, the pipeline `__call__` method should not take in `negative_prompt` or `negative_prompt_embeds` arguments, but we should still allow `encode_prompt` to prepare the default unconditional prompt `""` if `guidance_scale > 0.0`, and perform CFG as normal in the rest of the pipeline?

[My understanding is that the `diffusers` convention is to use the [Imagen](https://arxiv.org/pdf/2205.11487.pdf) CFG formulation $`\tilde{\epsilon}_\theta(z_t, c) = \omega'\epsilon_\theta(z_t, c) + (1 - \omega')\epsilon_\theta(z_t, \varnothing)`$ where $\omega' = \omega + 1$ and $\omega$ is the guidance scale used in the LCM paper, so I will probably change the guidance scale to follow the above definition.]

Reply:
Well, i think currently setting the negative prompt as False is correct. All we need to do is to change the comment for better clarification:  # LCM guided distillation currently only support unconditional prompt "", don't support other types of negative prompt.

Reply:
I have updated the comment to provide a (hopefully) better description about negative prompt support.

Reply:
Cool!

Reply:
> [My understanding is that the diffusers convention is to use the [Imagen](https://arxiv.org/pdf/2205.11487.pdf) CFG formulation 
 where 
 and 
 is the guidance scale used in the LCM paper, so I will probably change the guidance scale to follow the above definition.]

Indeed.